hazelcast/src/main/java/com/hazelcast/cp/CPMap.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cp;
+
+import com.hazelcast.core.DistributedObject;
+
+public interface CPMap<K, V> extends DistributedObject {
+    /**
```

Comment:
Add a performance not. In the past we had many users called map.put because that is the method in the java Map interface and returning a local reference is not a performance problem.

But with a remote call a put can be a wasteful if you don't care about the return value.